src/Microsoft.Azure.WebJobs.Extensions.ServiceBus/Listeners/ServiceBusScaleMonitor.cs

```diff
@@ -0,0 +1,337 @@
+
+            subscriptionRuntimeInfo = await _managementClient.Value.GetSubscriptionRuntimeInfoAsync(topicPath, subscriptionPath);
+            metrics.ActiveMessageCount = subscriptionRuntimeInfo.MessageCountDetails.ActiveMessageCount;
+            metrics.DeadLetterCount = subscriptionRuntimeInfo.MessageCountDetails.DeadLetterMessageCount;
+
+            // If partitioning is turned on, then Service Bus automatically partitions queues into 16 partitions
+            // See more information here: https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-partitioning#standard
+            topicDescription = await _managementClient.Value.GetTopicAsync(topicPath);
+            metrics.PartitionCount = topicDescription.EnablePartitioning ? 16 : 0;
+
+            return metrics;
+        }
+
+        public static ServiceBusTriggerMetrics CreateTriggerMetrics(ServiceBusMetricsData metrics, bool isUnauthorized, bool isListeningOnDeadLetterQueue)
+        {
+            long totalNewEventCount = 0;
+            TimeSpan eventQueueTime = TimeSpan.Zero;
+
+            if (metrics.Message != null)
+            {
+                eventQueueTime = DateTime.UtcNow.Subtract(metrics.Message.SystemProperties.EnqueuedTimeUtc);
+                totalNewEventCount = 1;
+            }
+
+            if (!isUnauthorized && ((!isListeningOnDeadLetterQueue && metrics.ActiveMessageCount > 0) || (isListeningOnDeadLetterQueue && metrics.DeadLetterCount > 0)))
+            {
+                totalNewEventCount = isListeningOnDeadLetterQueue ? metrics.DeadLetterCount : metrics.ActiveMessageCount;
+            }
+
+            return new ServiceBusTriggerMetrics
+            {
+                Timestamp = DateTime.UtcNow,
+                TotalNewEventCount = totalNewEventCount,
+                PartitionCount = metrics.PartitionCount,
+                EventQueueTime = eventQueueTime
+            };
+        }
+
+        ScaleStatus IScaleMonitor.GetScaleStatus(ScaleStatusContext context)
+        {
+            return GetScaleStatusCore(context.WorkerCount, context.Metrics?.Cast<ServiceBusTriggerMetrics>().ToArray());
+        }
+
+        public ScaleStatus GetScaleStatus(ScaleStatusContext<ServiceBusTriggerMetrics> context)
+        {
+            return GetScaleStatusCore(context.WorkerCount, context.Metrics?.ToArray());
+        }
+
+        private ScaleStatus GetScaleStatusCore(int workerCount, ServiceBusTriggerMetrics[] metrics)
+        {
+            ScaleStatus status = new ScaleStatus
+            {
+                Vote = ScaleVote.None
+            };
+
+            const int NumberOfSamplesToConsider = 5;
+
+            // Unable to determine the correct vote with no metrics.
+            if (metrics == null)
+            {
+                return status;
+            }
+
+            // We shouldn't assign more workers than there are partitions (Cosmos DB, Event Hub, Service Bus Queue/Topic)

```

Comment:
This code is specific to ServiceBus so remove the generalized comments for Cosmos/EventHub